src/Features/CSharp/Portable/TypeStyle/UseExplicitTypeCodeFixProvider.cs

```diff
@@ -79,7 +79,10 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             if (parensDesignation is null)
             {
                 var typeSymbol = semanticModel.GetTypeInfo(typeSyntax).ConvertedType;
-                var typeName = typeSymbol.GenerateTypeSyntax()
+
+                // We're going to be passed through the simplifier.  Tell it to not just convert
+                // this back to var (as that would defeat the purpose of this refactoring entirely).
+                var typeName = typeSymbol.GenerateTypeSyntax(allowVar: false)
```

Comment:
this same issue will likely affect you @jcouv .  If we have a refactoring to convert 'var' to a proper type, we'll likely want to put the "and do not convert this to var" bit on it so that the simplifier doesn't just undo that work.